mmengine/analysis/jit_analysis.py

```diff
@@ -0,0 +1,687 @@
+            dict[str, Counter(str)]:
+            The statistics for each submodule and each operator.
+            Grouped by submodule names, then by operator name.
+        """
+        stats = self._analyze()
+        return stats.counts
+
+    def by_module(self) -> typing.Counter[str]:
+        """Returns the statistics for all submodules, aggregated over all
+        operators.
+
+        Returns:
+            Counter(str): statistics counter grouped by submodule names
+        """
+        stats = self._analyze()
+        summed_counts = Counter()  # type: Counter
+        for mod, results in stats.counts.items():
+            summed_counts[mod] = sum(results.values())
+        return summed_counts
+
+    def unsupported_ops(self, module_name: str = '') -> typing.Counter[str]:
+        """Lists the number of operators that were encountered but unsupported
+        because no operator handle is available for them.
+
+        Does not include
+        operators that are explicitly ignored.
+
+        Args:
+            module_name (str) : The submodule to list unsupported ops.
+                Defaults to the entire model.
+
+        Returns:
+            Counter(str) : The number of occurrences each unsupported operator.
+        """
+        if self._stats is None:
+            raise RuntimeError('Analysis results should be computed '
+                               'before calling unsupported_ops()')
+        module_name = self.canonical_module_name(module_name)
+        return self._stats.unsupported_ops[module_name]  # pyre-fixme
+
+    def uncalled_modules(self) -> Set[str]:
+        """Returns a set of submodules that were never called during the trace
+        of the graph.
+
+        This may be because they were unused, or
+        because they were accessed via direct calls .forward() or with
+        other python methods. In the latter case, statistics will not be
+        attributed to the submodule, though the statistics will be included
+        in the parent module.
+
+        Returns:
+            set(str) : The set of submodule names that were never called
+            during the trace of the model.
+        """
+        stats = self._analyze()
+        return stats.uncalled_mods
+
+    def set_op_handle(self, *args,
+                      **kwargs: Optional[Handle]) -> 'JitModelAnalysis':
+        """Sets additional operator handles, or replaces existing ones.
+
+        Args:
+            args: (str, Handle) pairs of operator names and handles.
+            kwargs: mapping from operator names to handles.
+        If a handle is ``None``, the op will be explicitly ignored. Otherwise,
+        handle should be a function that calculates the desirable statistic
+        from an operator. The function must take two arguments, which are the
+        inputs and outputs of the operator, in the form of
+        ``list(torch._C.Value)``. The function should return a counter object
+        with per-operator statistics.
```

Comment:
```suggestion
        If a handle is ``None``, the op will be explicitly ignored. Otherwise,
        handle should be a function that calculates the desirable statistic
        from an operator. The function must take two arguments, which are the
        inputs and outputs of the operator, in the form of
        ``list(torch._C.Value)``. The function should return a counter object
        with per-operator statistics.
        
        Args:
            args: (str, Handle) pairs of operator names and handles.
            kwargs: mapping from operator names to handles.   
```